standalone/runtime/tests/test_redeem.rs

```diff
@@ -28,117 +28,369 @@ fn consume_to_be_replaced(vault: &mut CoreVaultData, amount_btc: u128) {
+            // use an iterative process to find the maximum redeem request amount.
+            let mut ret = redeemable_tokens + FeePallet::get_redeem_fee(redeemable_tokens).unwrap();
+
+            loop {
+                let actually_redeemed_tokens = ret - FeePallet::get_redeem_fee(ret).unwrap();
+                if actually_redeemed_tokens > redeemable_tokens {
+                    return ret - 1;
+                }
+                ret += 1;
+            }
+        }
+
+        #[test]
+        fn integration_test_request_redeem_at_capacity_succeeds() {
+            // Checked PRECONDITION: The vault’s `issuedTokens` MUST be at least `vault.toBeRedeemedTokens +
+            // burnedTokens`
+            test_with(|| {
+                let amount = calculate_vault_capacity();
+                assert_ok!(Call::Redeem(RedeemCall::request_redeem(
                     amount,
                     BtcAddress::default(),
                     account_of(VAULT)
                 ))
-                .dispatch(origin_of(account_of(USER))),
-                VaultRegistryError::InsufficientTokensCommitted
-            );
-        });
+                .dispatch(origin_of(account_of(USER))));
+
+                let redeem_id = assert_redeem_request_event();
+                let redeem = RedeemPallet::get_open_redeem_request_from_id(&redeem_id).unwrap();
+
+                assert_eq!(amount, redeem.fee + redeem.amount_btc + redeem.transfer_fee_btc);
+                assert_eq!(redeem.vault, account_of(VAULT));
+
+                assert_eq!(
+                    ParachainState::get(),
+                    ParachainState::default().with_changes(|user, vault, _, _| {
+                        vault.to_be_redeemed += redeem.amount_btc + redeem.transfer_fee_btc;
+                        user.free_tokens -= redeem.amount_btc + redeem.transfer_fee_btc + redeem.fee;
+                        user.locked_tokens += redeem.amount_btc + redeem.transfer_fee_btc + redeem.fee;
+                        consume_to_be_replaced(vault, redeem.amount_btc);
+                    })
+                );
+            });
+        }
+
+        #[test]
+        fn integration_test_request_redeem_above_capacity_fails() {
+            // Checked PRECONDITION: The vault’s `issuedTokens` MUST be at least `vault.toBeRedeemedTokens +
+            // burnedTokens`
+            test_with(|| {
+                let amount = calculate_vault_capacity() + 1;
+                assert_noop!(
+                    Call::Redeem(RedeemCall::request_redeem(
+                        amount,
+                        BtcAddress::default(),
+                        account_of(VAULT)
+                    ))
+                    .dispatch(origin_of(account_of(USER))),
+                    VaultRegistryError::InsufficientTokensCommitted
+                );
+            });
+        }
+
+        #[test]
+        fn integration_test_redeen_cannot_request_from_liquidated_vault() {
```

Comment:
Also in a couple of other places:
```suggestion
        fn integration_test_redeem_cannot_request_from_liquidated_vault() {
```